pmd-java/src/main/java/net/sourceforge/pmd/lang/java/symbols/internal/asm/GenericSigBase.java

```diff
@@ -163,7 +163,7 @@ protected boolean isGeneric() {
 
         @Override
         protected boolean postCondition() {
-            return (superItfs != null && superType != null || signature == null) && typeParameters != null;
+            return superItfs != null && (superType != null || signature == null) && typeParameters != null;
```

Comment:
I didn't figure out yet, why superItfs seems to end up as null. As far as I see, it is always initialized by the signature parser.

The superType can indeed be null - e.g. if we have java.lang.Object - and that can only happen, if signature == null. So I think, the parentheses are wrong. But I doubt it makes a big difference.

Reply:
superItfs ends up being null, if the doParse() call is aborted with a NPE - then we never set superItfs or superType.
I've also seen the postCondition failed when I reproduced the bug with Thread.sleep.